kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,325 @@
+    bottomSheetBehavior.peekHeight = bottomSheet.header.height
+    (mainContainer.layoutParams as CoordinatorLayout.LayoutParams).bottomMargin =
+      bottomSheet.header.height
+
+    // animate the arrow when the bottom sheet slides
+    bottomSheetBehavior.addBottomSheetCallback(object : BottomSheetCallback() {
+      override fun onSlide(bottomSheet: View, slideOffset: Float) {
+        bottomSheet.header.imageView.rotation = slideOffset * 180f
+      }
+
+      override fun onStateChanged(bottomSheet: View, newState: Int) {
+        bottomSheet.header.imageView.rotation = when (newState) {
+          BottomSheetBehavior.STATE_EXPANDED -> 180f
+          else -> 0f
+        }
+      }
+    })
+
+    // expand or collapse the bottom sheet when the header is clicked
+    bottomSheet.header.setOnClickListener {
+      bottomSheetBehavior.state = when (bottomSheetBehavior.state) {
+        BottomSheetBehavior.STATE_COLLAPSED -> BottomSheetBehavior.STATE_EXPANDED
+        else -> BottomSheetBehavior.STATE_COLLAPSED
+      }
+    }
+
+    bottomSheet.directionsListView.apply {
+      // Set the adapter for the list view
+      adapter = ArrayAdapter(
+        this@MainActivity,
+        android.R.layout.simple_list_item_1,
+        directions.map { it.directionText }
+      )
+
+      // when the user taps a maneuver, set the viewpoint to that portion of the route
+      onItemClickListener =
+        AdapterView.OnItemClickListener { _, _, position, _ ->
+          // remove any graphics that are not the two stops and the route graphic
+          if (graphicsOverlay.graphics.size > 3) {
+            graphicsOverlay.graphics.removeAt(graphicsOverlay.graphics.size - 1)
+          }
+
+          // set the viewpoint to the selected maneuver
+          val geometry = directions[position].geometry
+          mapView.setViewpointAsync(
+            Viewpoint(geometry.extent, 20.0),
+            1f
+          )
+
+          // create a graphic with a symbol for the maneuver and add it to the graphics overlay
+          val selectedRouteSymbol = SimpleLineSymbol(
+            SimpleLineSymbol.Style.SOLID,
+            Color.GREEN, 5f
+          )
+          Graphic(geometry, selectedRouteSymbol).also {
+            graphicsOverlay.graphics.add(it)
+          }
+
+          // collapse the bottom sheet
+          bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
+        }
+    }
+  }
+
+  /** Create a progress dialog box for tracking the route task.
+   *
+   * @param routeTask the route task progress to be tracked
+   * @return an AlertDialog set with the dialog layout view
+   */
+  private fun createProgressDialog(routeTask: RouteTask) =
+    AlertDialog.Builder(this@MainActivity)
+      .setTitle("Solving route...")
+      // provide a cancel button on the dialog
+      .setNegativeButton("Cancel") { _, _ ->
+        routeTask.cancelLoad()
+      }
+      .setCancelable(false)
+      .setView(R.layout.dialog_layout)
+      .create()
```

Comment:
I don't think the calculation warrant it's own progress dialog. Maybe just overlay an invisible indeterminate progress indicator in the middle of the map view and toggle it's visibility based on a done loading listener

Reply:
This is a good idea!